CredentialsManager/Models/Credentials.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+
+namespace CredentialsManager.Models
+{
+  public class Credentials : IEquatable<Credentials>, IComparable<Credentials>, IComparable
+  {
+    public string UserName { get; }
+    public string UserPassword { get; }
+
+    public Credentials(string userName, string userPassword)
+    {
+      if (string.IsNullOrWhiteSpace(userName))
+      {
+        throw new ArgumentException("Value can not be null or empty", nameof(userName));
+      }
+
+      if (string.IsNullOrWhiteSpace(userPassword))
+      {
+        throw new ArgumentException("Value can not be null or empty", nameof(userName));
+      }
+      
+      UserName = userName;
+      UserPassword = userPassword;
+    }
+
+    public static bool TryParse(string line, string delimiter, out Credentials credentials)
+    {
+      credentials = default;
+      const int validRowLength = 2;
+
+      if (string.IsNullOrWhiteSpace(line)) return false;
+
+      var row = line.Split(delimiter);
+      if (row.Length != validRowLength) return false;
+
+      var userName = row[0].Trim();
+      var userPassword = row[1].Trim();
+
+      if (string.IsNullOrWhiteSpace(userName) || string.IsNullOrWhiteSpace(userPassword))
+      {
+        return false;
+      }
+
+      credentials = new Credentials(userName, userPassword);
+
+      return true;
+    }
+
+    public override string ToString()
+    {
+      return $"{UserName},{UserPassword}";
+    }
+
+    public bool Equals(Credentials? other)
```

Comment:
I don't think the `?` is necessary here?  But I might be wrong.  I'm not up to speed on this operator.  Maybe it's needed for one of the interfaces.

Reply:
At the time you were reviewing my code I was in process of converting the class to a struct as per Kai's suggestion and so I did. I think it is required for classes but not for structs.
The commit is this one [4b30e14](https://github.com/csinn/CSharp-From-Zero-To-Hero-v2/pull/43/commits/4b30e147718194bba663095be2c2d4fa23a90759)